_includes/tutorials/produce-consume-lang/scala/markup/dev/5_0-produce-narrative.adoc

```diff
@@ -0,0 +1,47 @@
+The following Scala class will be the entry point of the producer application.
+At `src/main/scala/io/confluent/developer/Producer.scala` add:
+
++++++
+<pre class="snippet"><code class="java">{%
+    include_raw tutorials/produce-consume-lang/scala/code/src/main/scala/io/confluent/developer/Producer.scala
+%}</code></pre>
++++++
+
+Lets's describe the keys pieces of this program.
+
+- Reminder: we use an externalised config and this app loads its config from the _producer_ bloc
+of the `application.conf`
+
+A `Producer#produce` function covers most of the record production.
+
+- `Producer#produce` takes a
+`https://kafka.apache.org/25/javadoc/index.html?org/apache/kafka/clients/producer/KafkaProducer.html[KafkaProducer]`,
+a topic name, and an instance of book to send into a Kafka topic.
+
+- `Producer#produce` wraps our book with a `ProducerRecord[K, V]`, that's where attache the topic name to the book.
+
+- `Producer#produce` call the `KafkaProducer#send` method and gets back Java future tight to the broker acknowledgement.
+
+- `Producer#produce` use a
+`https://kafka.apache.org/25/javadoc/index.html?org/apache/kafka/clients/producer/KafkaProducer.html[CallBack]`
+to confirm the future completion and the record write.
```

Comment:
Great details on the code. We've recently added some additional asciidoctor support
.
Could you put the `Producer.produce`  method in a code block with callouts? 

You probably don't need this but [here's an example](https://github.com/confluentinc/kafka-tutorials/blob/master/_includes/tutorials/serialization/kstreams/markup/dev/make-topology.adoc)